src/main/java/decodex/storage/Storage.java

```diff
@@ -107,8 +108,14 @@ public void loadRecipesFromDirectory(ModuleManager moduleManager, RecipeManager
      *
      * @return The list of recipe File objects.
      */
-    private File[] getAllRecipeFiles() {
+    private File[] getAllRecipeFiles() throws StorageException {
         File recipeDirectory = new File(DEFAULT_RECIPE_DIRECTORY);
+
+        if (!recipeDirectory.isDirectory()) {
+            // throw error if not a valid directory.
+            throw new StorageException(ErrorMessages.INVALID_RECIPE_DIRECTORY_FILETYPE);
+        }

```

Comment:
This is on me but when I think about it, perhaps checking if its NULL is better because this will handle if the recipeDirectory is not a directory as well as handle i/o issues (maybe issues when the os listing the files)

Reply:
Can consider checking existence with `recipeDirectory.exists()` as well.